Mathlib/Analysis/Meromorphic/LeadCoefficient.lean

```diff
@@ -0,0 +1,250 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+import Mathlib.Analysis.Meromorphic.Order
+
+/-!
+# The Leading Coefficient of a Meromorphic Function
+
+This file defines the leading coefficient of a meromorphic function. If `f` is meromorphic at a
+point `x`, the leading coefficient is defined as the (unique!) value `g x` for a presentation of `f`
+in the form `(z - x) ^ order • g z` with `g` analytic at `x`.
+
+The lemma `leadCoefficient_eq_limit` expresses the leading coefficient as a limit.
+-/
+
+variable
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  {f g : 𝕜 → E} {x : 𝕜} {n : ℤ}
+
+open Filter Topology
+
+namespace MeromorphicAt
+
+variable (f x) in
+/--
+If `f` is meromorphic of finite order at a point `x`, the leading coefficient is defined as the
+(unique!) value `g x` for a presentation of `f` in the form `(z - x) ^ order • g z` with `g`
+analytic at `x`. In all other cases, the leading coefficient is defined to be zero.
+-/
+noncomputable def leadCoefficient : E := by
```

Comment:
A definition should not use a tactic like `rw`. Could you instead define it using `by_cases h₁ : MeromorphicAt f x`, to avoid the rewrite?

Also, I think you could go for `leadCoeff`, which is shorter but not ambiguous at all.